NachoNight/auth-api
# 4/editing users

Issue #4 

TL;DR - What does this implement?
---------------------------------
Mailing system, password recovery, caching, account, email change and password change verification.

How to Test
-----------
Start the Node server using `npm start`,
then run `npm test`.

Worries / Concerns
------------------
Again, security.
I just remembered one weak point of JWT authentication that I didn't list as a concern in my previous PR. If a user changes it's credentials, removes it's account or changes the account in question in any way, the token will still be valid for use until it expires for some endpoints. The next thing to be worked on should most certainly be switching over to OAuth. 

EDIT: Another concern is caching. Would it be better to run a Redis instance for caching, or should we keep running the current cache setup?

Checklist
---------

 - [X] Is this is code change?
 
ESLint Output
-------------

```
[Info  - 5:48:26 PM] ESLint server stopped.
[Info  - 5:48:26 PM] ESLint server running in node v10.11.0
[Info  - 5:48:26 PM] ESLint server is running.
[Info  - 5:48:27 PM] ESLint library loaded from: /home/denis/code/nacho-night-auth-api/node_modules/eslint/lib/api.js
Warning: React version was set to "detect" in eslint-plugin-react settings, but the "react" package is not installed. Assuming latest React version for linting
```

Reviewers
---------
@gmemstr
